cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -884,6 +885,101 @@ public void deleteFlowQueues(HttpRequest request, HttpResponder responder,
     }
   }
 
+  @POST
+  @Path("/apps/{app-id}/workflows/{workflow-name}/{run-id}/suspend")
```

Comment:
Since workflows have some specific APIs (suspend, resume, and in future ``ExecutionContext``) should we extract them out to a ``WorkflowHttpHandler`` that extends ``ProgramLifecycleHttpHandler``? ``ProgramLifecycleHttpHandler`` is already becoming large, and these APIs are very specific to only 1 type of program - workflow.